source/extensions/filters/common/lua/wrappers.h

```diff
@@ -274,7 +308,11 @@ class SslConnectionWrapper : public BaseLuaObject<SslConnectionWrapper> {
 
   // TODO(dio): Add luaX509Extension if required, since currently it is used out of tree.
 
+  // Envoy::Lua::BaseLuaObject
+  void onMarkDead() override { parsed_subject_peer_certificate_.reset(); }
+
   const Ssl::ConnectionInfo& connection_info_;
+  LuaDeathRef<ParsedX509NameWrapper> parsed_subject_peer_certificate_;
```

Comment:
Why does this need to be LuaDeathRef, and `connection_info_` above is a bare reference?

Reply:
because the connection_info_ is an internal object used by code, while the parsed_subject_peer_certificate_ is actually a cached lua object will be returned to user. you can see similar usage in line 340 for ConnectionWrapper.